python_jsonschema_objects/classbuilder.py

```diff
@@ -24,16 +24,18 @@ class ProtocolBase( collections.MutableMapping):
     }
 
     def as_dict(self):
-      out = {}
-      for prop in self:
-          propval = getattr(self, prop)
+        out = {}
+        for prop in self:
+            propval = getattr(self, prop)
 
-          if isinstance(propval, list):
+            if isinstance(propval, list):
               out[prop] = [x.as_dict() for x in propval]
-          elif propval is not None:
+            elif not isinstance(propval, self):
```

Comment:
methinks this should be `isinstance(propval, ProtocolBase)`, but I don't know what's causing the other test error.